File: backend/src/post/dto/comment.dto.ts
export type CommentInfos = {
    id: number;
    content: string;
    creationDate: Date;
    updatedAt?: Date;
    isEdited: boolean;
    nbLikes: number;
    nbReplies: number;
    userId: number;
    username: string;
    avatar: string;
    parentId?: number;
}

File: frontend/app/src/main/java/com/example/myapplication/dto/comment/CommentDto.kt
package com.example.myapplication.dto.comment

data class CommentDto(
    val id: Int,
    val content: String,
    val creationDate: String,
    val updatedAt: String?,
    val isEdited: Boolean,
    val nbLikes: Int,
    val nbReplies: Int,
    val userId: Int,
    val username: String,
    val avatar: String,
    val parentId: Int?,
    val isYours: Boolean,
    val isLiked: Boolean
)

data class CreateCommentDto(
    val content: String,
    val parentId: Int? = null
)

data class UpdateCommentDto(
    val content: String
)
